tests/phpunit/tests/functions/wpCheckFiletype.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+/**
+ * Test the wp_upload_bits function
+ *
+ * @group Functions
+ * @group Upload
+ * @covers ::wp_check_filetype
+ */
+class Test_wp_check_filetype extends WP_UnitTestCase {
+
+	/**
+	 * @ticket 57151
+	 * @dataProvider wp_check_filetype_dataset
+	 */
+	function test_wp_check_filetypes( $filename, $mines, $expected ) {
+		$this->assertSame( $expected, wp_check_filetype( $filename, $mines ) );
+	}
+
+	function wp_check_filetype_dataset() {
+		return array(
+			'default'     => array(
+				'filename' => 'canola.jpg',
+				'mines'    => null,
+				'expected' => array(
+					'ext'  => 'jpg',
+					'type' => 'image/jpeg',
+				),
+			),
+			'short_mines' => array(
```

Comment:
```suggestion
			'.jpg and jpg|jpeg|jpe mimetypes'    => array(
```
I'd also suggest these additional datasets:
1. A `.jpeg` file with the same value for `$mimes` as this dataset.
2. A `.jpe` file with the same value for `$mimes` as this dataset.
3. A `.JPG` file (uppercase) to ensure the regex in `wp_check_filetype()` is always case-insensitive.